### PR TITLE
Created testcase for adding milestones with space in name #2051  

### PR DESCRIPTION
**Pull request by loro2**
PR for Issue #2051 unrecoverable failure Type: "Job milestoned".
Reproduced the problem with a unit test and added puzzle for further development.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#2189](https://github.com/zerocracy/farm/issues/2189) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#2189](https://github.com/zerocracy/farm/issues/2189) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/5ef49021-0ffb-49e6-ba10-1ebc59e3a7e5); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/com/zerocracy/pm/time/MilestonesTest.java [hunk 1]
```diff
@@ -45,4 +46,29 @@ public void iterateAllAddedMilestones() throws Exception {
             Matchers.hasItem(Matchers.equalTo(target))
         );
     }
+
+    /**
+     * Puzzle from Issue #2051: unrecoverable failure Type: "Job milestoned".
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pm/time/MilestonesTest.java]: @loro2 Let's change this "Test if milestones with spaces can be added". It's a better description of what the bug was and the scenario you are trying to reproduce. There is also no need to reference the puzzle here - you can do that in the commit message instead.

Comment (loro2) [reply src/test/java/com/zerocracy/pm/time/MilestonesTest.java]: Done.

**Comment by loro2** [comment]
I didn't break the build, master is broken since 3 days ago.

**Comment by carlosmiranda** [comment]
@rultor good to merge

**Comment by g4s8** [comment]
@loro2 you also can submit a ticket to https://github.com/zerocracy/datum/issues/new to fix milestone regex pattern (you'll get 15 reputation in that case, otherwise next developer of your todo will do that)

**Comment by g4s8** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@g4s8 @loro2 any update on the build?

**Comment by g4s8** [comment]
@carlosmiranda the issue should be fixed in master #2200, I'm waiting @loro2 to merge master into this branch to restart the build

**Comment by loro2** [comment]
@carlosmiranda @g4s8 merged

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (23 days), architects (@g4s8) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#2189](https://github.com/zerocracy/farm/issues/2189) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by sereshqua** [comment]
@loro2 please make sure you start all your comments with the name of the user they are referred to, see: [1](https://github.com/zerocracy/farm/pull/2189#discussion_r341033791), [2](https://github.com/zerocracy/farm/pull/2189#issuecomment-548285129)

@carlosmiranda please make sure you find at least 3 issues, which also are NOT just cosmetic, during the next CR

**Comment by carlosmiranda** [comment]
@sereshqua confirmed

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by loro2** [comment]
@sereshqua confirmed

**Pull request merged**